buildSrc/src/main/java/org/elasticsearch/gradle/internal/precommit/InternalPrecommitTasks.java

```diff
@@ -21,7 +22,12 @@
      */
     public static void create(Project project, boolean includeDependencyLicenses) {
         PrecommitTasks.create(project);
-
+        if (BuildParams.isInternal() && project.getPath().equals(":libs:elasticsearch-core") == false) {
```

Comment:
Why are we checking `isInternal()` here? This is an internal-only plugin so shouldn't be used by external users, right?

Reply:
There are still some BuilParams.isInternal() usages left in our internal plugins. The main reason I didn't rip them out as part of this PR is that removing them makes a huge set of our plugin functional tests fail as they would require a more sophisticated test project structure. It's related to the problem of declaring default (project) dependencies in plugins as we discussed few days ago which makes testing our plugins way harder.

I will create some follow up tickets to track and sort this (the removal of BuildParams.isInternal()) and the better handling of dependencies out in our codebase.

Reply:
Roger that. The test fallout makes sense.